tests/test_views.py

```diff
@@ -0,0 +1,1050 @@
+            ],
+            'json_data': None,
+        }
+
+        for key in expected:
+            assert str(response.context[key]) == str(expected[key])
+
+    @pytest.mark.parametrize('reg_required, data, expected_errors', [
+        (
+            False,
+            {'scope': 2},
+            {'scope': [u'Select a valid choice. 2 is not one of the available choices.']}
+        ),
+        (
+            False,
+            {'scope': 1, 'nominator_name': 'John Doe', 'nominator_institution': 'UNT'},
+            {
+                'nominator_email': (
+                    u'You must provide name, institution, and '
+                    'email to affiliate your name or institution with nominations. Leave '
+                    'all "Information About ' 'You" fields blank to remain anonymous.'
+                )
+            }
+        ),
+        (
+            True,
+            {'scope': 1, 'nominator_name': 'John Doe', 'nominator_institution': 'UNT'},
+            {'nominator_email': u'This field is required.'}
+        )
+    ])
+    def test_context_with_post_and_errors(self, client, reg_required, data, expected_errors):
+        entity = 'www.example.com'
+        project = factories.ProjectFactory(registration_required=reg_required)
+        factories.URLFactory(url_project=project, entity=entity)
+        response = client.post(
+            reverse('url_listing', args=[project.project_slug, entity]),
+            data
+        )
+
+        assert response.context['form_errors'] == expected_errors
+        assert response.context['summary_list'] is None
+        assert response.context['scope_form'].is_valid() is False
+        assert sorted(json.loads(response.context['json_data'])) == sorted(
+            [[key, [str(data[key])]] for key in data])
+
+    def test_context_with_get(self, client):
+        entity = u'http://www.example.com'
+        entity_surt = u'http://(com,example,www,)'
+        project = factories.ProjectFactory()
+        factories.ProjectMetadataFactory(project=project)
+        factories.URLFactory(url_project=project, entity=entity)
+        factories.SURTFactory(
+            url_project=project,
+            entity=entity,
+            value=entity_surt
+        )
+        factories.SURTFactory.create_batch(
+            1,
+            url_project=project,
+            entity=u'{}/main/page'.format(entity),
+            value=u'{}/main/page'.format(entity_surt)
+        )
+
+        expected_context = {
+            'scope_form': views.ScopeForm(),
+            'form_errors': None,
+            'summary_list': None,
+            'json_data': None
+        }
+        response = client.get(reverse('url_listing', args=[project.project_slug, entity]))
+
+        for key in expected_context:
+            assert str(response.context[key]) == str(expected_context[key])
+
+
+class TestUrlSurt():
+
+    def test_status_ok(self, rf):
```

Comment:
I think in this case the response if a slug doesn't exist and a surt doesn't exist are different. I think the 404 raised by a non-existent slug is coming from `get_project`. A non-existent surt looks like it still returns a 200--though perhaps in the future if we want it to be a 404, we should mark this as an expected failure.